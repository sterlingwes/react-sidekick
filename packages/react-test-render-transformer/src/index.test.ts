import { transform } from "./index";
import { rnGradientJSONFixture } from "./fixtures/react-native-to-json";

describe("test renderer transformer", () => {
  describe("simple RN hierarchy", () => {
    it("should return JS string with one element", () => {
      const jsString = transform(rnGradientJSONFixture);
      expect(jsString).toMatchSnapshot();
    });
  });
});
