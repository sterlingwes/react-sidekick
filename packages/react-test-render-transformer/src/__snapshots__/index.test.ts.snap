// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test renderer transformer simple RN hierarchy should return JS string with one element 1`] = `"g.react.createElement(g.react.Fragment, null, g.react.createElement(g.rn.RCTSafeAreaView, {\\"emulateUnlessSupported\\":true,\\"style\\":{\\"flex\\":1}}, g.react.createElement(g.rn.View, {\\"style\\":[{\\"flex\\":1,\\"backgroundColor\\":\\"#F9F8F7\\",\\"paddingTop\\":30},null]}, g.react.createElement(g.rn.View, {\\"style\\":{\\"flex\\":1}}, g.react.createElement(g.rn.View, {\\"style\\":{\\"marginHorizontal\\":20}}, g.react.createElement(g.rn.Text, {\\"style\\":[{\\"fontFamily\\":\\"ComicSans\\"},{\\"fontSize\\":26,\\"lineHeight\\":36},[[{\\"color\\":\\"#403E3D\\"},[{\\"lineHeight\\":36,\\"letterSpacing\\":0.2,\\"color\\":\\"#403E3D\\"},{\\"color\\":\\"#403E3D\\",\\"marginBottom\\":20}]]],null],\\"adjustsFontSizeToFit\\":false,\\"minimumFontScale\\":null}, \\"This is a title\\"),g.react.createElement(g.rn.Text, {\\"style\\":[{\\"fontFamily\\":\\"ComicSans\\"},{\\"fontSize\\":14,\\"lineHeight\\":19},[[{\\"color\\":\\"#403E3D\\"},[{\\"fontSize\\":16,\\"lineHeight\\":24,\\"color\\":\\"#73706E\\"},{\\"color\\":\\"#403E3D\\"}]]],null]}, \\"This is some body text.\\")),g.react.createElement(g.rn.View, {\\"style\\":[{\\"flex\\":1,\\"alignItems\\":\\"center\\",\\"justifyContent\\":\\"center\\"},null]}, g.react.createElement(g.rn.Image, {\\"style\\":{\\"width\\":200,\\"height\\":200},\\"source\\":g.placeholderImage}, undefined))),g.react.createElement(g.rn.View, {\\"style\\":[{\\"paddingHorizontal\\":20,\\"paddingVertical\\":10,\\"alignItems\\":\\"center\\"},{\\"backgroundColor\\":\\"#F9F8F7\\"}]}, g.react.createElement(g.rn.View, {\\"accessible\\":true,\\"accessibilityRole\\":\\"button\\",\\"accessibilityState\\":{\\"disabled\\":false},\\"style\\":{\\"borderRadius\\":20,\\"flexDirection\\":\\"row\\",\\"opacity\\":1},\\"testID\\":\\"ModalButtonPrimary\\",\\"isTVSelectable\\":true,\\"focusable\\":true}, g.react.createElement(g.rn.BVLinearGradient, {\\"style\\":[{\\"borderRadius\\":20,\\"flexDirection\\":\\"row\\",\\"justifyContent\\":\\"center\\",\\"alignItems\\":\\"center\\",\\"flex\\":1,\\"backgroundColor\\":\\"#403e3d\\"}],\\"startPoint\\":{\\"x\\":0.5,\\"y\\":0},\\"endPoint\\":{\\"x\\":0.5,\\"y\\":1},\\"colors\\":[4294757485,4294688341],\\"locations\\":null}, g.react.createElement(g.rn.Text, {\\"style\\":[{\\"fontFamily\\":\\"ComicSans\\"},{\\"fontSize\\":16,\\"lineHeight\\":22},[[{\\"color\\":\\"#403E3D\\"},[{\\"zIndex\\":2,\\"textAlign\\":\\"center\\",\\"color\\":\\"#403E3D\\",\\"flex\\":6,\\"marginVertical\\":10,\\"marginHorizontal\\":20}]]],null]}, \\"Login\\"),g.react.createElement(g.rn.View, {\\"style\\":{\\"position\\":\\"absolute\\",\\"top\\":0,\\"left\\":0,\\"bottom\\":0,\\"right\\":0,\\"backgroundColor\\":\\"#1A1A19\\",\\"borderRadius\\":20,\\"opacity\\":0},\\"pointerEvents\\":\\"none\\"}, undefined))),g.react.createElement(g.rn.View, {\\"accessible\\":true,\\"accessibilityRole\\":\\"button\\",\\"accessibilityState\\":{\\"disabled\\":false},\\"style\\":{\\"borderRadius\\":20,\\"flexDirection\\":\\"row\\",\\"opacity\\":1},\\"testID\\":\\"ModalButtonSecondary\\",\\"isTVSelectable\\":true,\\"focusable\\":true}, g.react.createElement(g.rn.View, {\\"style\\":[{\\"borderRadius\\":25,\\"flexDirection\\":\\"row\\",\\"justifyContent\\":\\"center\\",\\"alignItems\\":\\"center\\",\\"flex\\":1,\\"backgroundColor\\":\\"transparent\\"},{\\"marginTop\\":10}],\\"colors\\":[\\"transparent\\",\\"transparent\\"]}, g.react.createElement(g.rn.Text, {\\"style\\":[{\\"fontFamily\\":\\"ComicSans\\"},{\\"fontSize\\":14,\\"lineHeight\\":19},[[{\\"color\\":\\"#403E3D\\"},[{\\"zIndex\\":2,\\"textAlign\\":\\"center\\",\\"color\\":\\"#73706E\\",\\"flex\\":6,\\"marginVertical\\":10,\\"marginHorizontal\\":20}]]],null]}, \\"Learn more\\"),g.react.createElement(g.rn.View, {\\"style\\":{\\"position\\":\\"absolute\\",\\"top\\":0,\\"left\\":0,\\"bottom\\":0,\\"right\\":0,\\"backgroundColor\\":\\"#1A1A19\\",\\"borderRadius\\":25,\\"opacity\\":0},\\"pointerEvents\\":\\"none\\"}, undefined)))))))"`;
